### ISSUE TITLE
[Request] Support app icon PNG data command

### ISSUE THREAD

**Issue by JJdeGroot** [facebook/idb#849]
### Description

I would like to submit a feature request to support the retrieval of app icons.

### Solution

In file: `FBSpringboardServicesClient.m` add method `getAppIcon:(NSString *)bundleId`

Command:
```objc
[self.connection sendAndReceiveMessage:@{@"command": @"getIconPNGData", @"bundleId": bundleId, @"formatVersion": @"2"} error:error];
```
This command should return the icon PNG data for the app with the specified bundle id.

In file: `app.py` add class `AppIconCommand(ClientCommand)` with name `--app-icon`, taking `bundle-id` as argument.

Extra: expand `AppListCommand` with argument `--icons` to retrieve icons for each app.

I've tried adding this myself, but can't manage to get the companion project to compile.

**Comment by answer-huang** [comment]
you can use following cmd get app location
```
xcrun simctl get_app_container${device_id} ${bundle_id} app
```
and find icon in the app container.

**Comment by JJdeGroot** [comment]
@answer-huang this only works for iOS simulators.

For iOS 17+ we now have `devicectl`, but the app icon command seems to be broken: https://developer.apple.com/forums/thread/738793

**Comment by answer-huang** [comment]
> @answer-huang this only works for iOS simulators.
> 
> For iOS 17+ we now have `devicectl`, but the app icon command seems to be broken: https://developer.apple.com/forums/thread/738793

`devicectl` works fine in my computer with:
MacOS: 14.0
iPhone: 17.0.3
Xcode: 15.0

**Comment by JJdeGroot** [comment]
@answer-huang yeah devicectl works fine, just the app icon command was broken before macOS 14.x - but works now.
